src/components/file-dropzone/file-dropzone.tsx

```diff
@@ -0,0 +1,162 @@
+ * by dragging and dropping them into the specified area.
+ *
+ * After receiving the files, the component emits a `filesSelecgted` event. For unsupported
+ * files (specified with the `accept` prop) a `filesRejected` event will be emitted.
+ *
+ * The event detail would be an array of `FileInfo` objects,
+ * each representing a file dropped into the dropzone.
+ *
+ * @exampleComponent limel-example-file-dropzone
+ * @exampleComponent limel-example-file-dropzone-type-filtering
+ * @private
+ */
+@Component({
+    tag: 'limel-file-dropzone',
+    shadow: true,
+    styleUrl: 'file-dropzone.scss',
+})
+export class FileDropzone {
+    /**
+     * Specifies the types of files that the dropzone will accept. By default, all file types are accepted.
+     *
+     * For media files, formats can be specified using: `audio/*`, `video/*`, `image/*`.
+     * Unique file type specifiers can also be used, for example: `.jpg`, `.pdf`.
+     * A comma-separated list of file extensions or MIME types is also acceptable, e.g., `image/png, image/jpeg` or
+     * `.png, .jpg, .jpeg`.
+     *
+     * @see [HTML attribute: accept](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept) for more
+     * details.
+     */
+    @Prop({ reflect: true })
+    public accept: string = '*';
+
+    /**
+     * Set to `true` to disable the file dropzone.
+     */
+    @Prop()
+    public disabled: boolean = false;
+
+    /**
+     * Text to be displayed when the user is dragging a file over the dropzone.
+     */
+    @Prop()
+    public onDragLayoutText: string = '';
+
+    /**
+     * Icon to be displayed when the user is dragging a file over the dropzone.
+     */
+    @Prop()
+    public onDropLayoutIcon: string = 'upload_2';
+
+    /**
+     * Sets to true when there is a file to drop
+     */
+    @State()
+    private hasFileToDrop: boolean = false;
+
+    /**
+     * Emitted when files are selected
+     */
+    @Event()
+    filesSelected: EventEmitter<FileInfo[]>;
+
+    /**
+     * Emitted when files are selected but do not conform with the `accept` property specifications.
+     * This can happen when the file types or formats of the selected files are not among the ones allowed by the dropzone,
+     * as defined by the `accept` property.
+     *
+     * @see {@link accept} for details on how to specify acceptable file types.
+     */
+    @Event()
+    filesRejected: EventEmitter<FileInfo[]>;
+
+    public render() {
+        return (
+            <Host
+                onDrop={this.handleDrop}
+                onDragOver={this.handleDragOver}
+                onDragLeave={this.handleDragLeave}
+            >
+                <slot />
+                {this.renderOnDragLayout()}
+            </Host>
+        );
+    }
+
+    private renderOnDragLayout = () => {
+        if (this.hasFileToDrop) {
+            return (
+                <div class="has-file-to-drop">
+                    <limel-icon
+                        class="dropzone-icon"
```

Comment:
```suggestion
                        class="icon"
```

The CSS is already scoped and the internal elements are isolated. There is no need for such specificity in web components.